cli/test/utils/FakeApi.js

```diff
@@ -1,15 +1,22 @@
 module.exports = class FakeApi {
-  constructor(response, status) {
-    this.desiredResponse = response;
-    this.desiredStatus = status;
+  constructor(mockValueObj) {
+    Object.keys(mockValueObj).forEach(
```

Comment:
I opted to rewrite this so that tests can build the class endpoints they like. So, given a key like, `whoami` it will bind a function to be invoked to the `FakeApi` class. 

While this doesn't document the api as much, it's much more flexible and handles things like `publish` where multiple responses need to be built.